src/base/net/downloadhandler.cpp

```diff
@@ -92,8 +92,10 @@ void DownloadHandler::processFinishedDownload()
             // Success
             QByteArray replyData = m_reply->readAll();
             if (m_reply->rawHeader("Content-Encoding") == "gzip") {
-                // uncompress gzip reply
-                Utils::Gzip::uncompress(replyData, replyData);
+                // decompress gzip reply
+                QByteArray tmp;
```

Comment:
IMO, the best method signature in this case is `QByteArray decompress(const QByteArray &data, bool *ok = nullptr)`. Similar for compress.